analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -126,8 +126,25 @@ class NuGetSupport {
     private fun getAllPackageData(id: Identifier): NuGetAllPackageData {
         // Note: The package name in the URL is case-sensitive and must be lower-case!
         val lowerId = id.name.toLowerCase()
-        val dataUrl = "$registrationsBaseUrl/$lowerId/${id.version}.json"
-        val data = runBlocking { mapFromUrl<PackageData>(JSON_MAPPER, dataUrl) }
+
+        var data: PackageData? = null
+        for (baseUrl in registrationsBaseUrls) {
+            val dataUrl = "$baseUrl/$lowerId/${id.version}.json"
+            data = try {
+                runBlocking { mapFromUrl<PackageData>(JSON_MAPPER, dataUrl) }
+            } catch (e: IOException) {
+                log.debug("Failed to retrieve package from $baseUrl")
+                null
+            }
+
+            if (data != null) {
+                break
+            }
+        }
+        if (data == null) {
+            throw IOException("Failed to retrieve package data for $lowerId from any registrationsBaseUrls: " +
+                    "$registrationsBaseUrls")
+        }
```

Comment:
How about this instead:

        val data = registrationsBaseUrls.asSequence().mapNotNull { baseUrl ->
            try {
                val dataUrl = "$baseUrl/$lowerId/${id.version}.json"
                runBlocking { mapFromUrl<PackageData>(JSON_MAPPER, dataUrl) }
            } catch (e: IOException) {
                log.debug { "Failed to retrieve package from $baseUrl". }
                null
            }
        }.firstOrNull()
            ?: throw IOException("Failed to retrieve package data for '$lowerId' from any of $registrationsBaseUrls.")

The "conversion" to a sequence results in not all elements being mapped a prior, but only "on demand" until the first non-null element (or none) is found.